packages/lesmis/src/components/SearchBar/utils.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ *
+ */
+import React from 'react';
+import { ReactComponent as Province } from '../icons/province.svg';
+import { ReactComponent as District } from '../icons/district.svg';
+import { ReactComponent as School } from '../icons/school.svg';
+
+/**
+ * Get the display icon for search results
+ */
+export const getPlaceIcon = type => {
+  switch (type) {
+    case 'Province':
+      return <Province />;
+    case 'District':
+      return <District />;
+    case 'Village':
+      return <School />;
+    default:
+      return <District />;
+  }
+};
+
+/**
+ * Get the display text for search results
+ */
+export const getOptionText = (option, countryHeirarchy, hierarchy = []) => {
+  const { type, name, parent } = option;
+
+  const newHierarchy = [...hierarchy, name];
+
+  if (type === 'Country') {
+    return newHierarchy.join(', ');
+  }
+
+  const parentOrgUnit = countryHeirarchy.find(entity => entity.organisationUnitCode === parent);
```

Comment:
As an optimisation, it could be worth passing in an object or Map that has entities keyed by code. I think we'll end up with this being useful in other places, so it could be worth creating it from the array right back in the `useCountryHierarchyData` hook so that consumers can use either data structure
```
const { entities, entityByCode } = useCountryHierarchyData;
```

Reply:
I had a play with doing this and if it's ok with you I've decided to not do it but keep it in mind. I'm weary of adding another variable to memory which is in the order of 3mb when it seems to be performing well as it is and doing the find in a couple of places is very easy. Just going for the library object is also not an option because the material autocomplete component expects an array.

Reply:
Fair enough, no need to optimise early if it's performing fine :-)

There would be other optimisation options as well, like caching the result so that if the same `option` was passed in, the result was returned immediately rather than looking through again (because at the moment the same results might be running through this loop each time the search term changes)